drivers/serial/uart_nrf_sw_lpuart.c

```diff
@@ -567,6 +608,206 @@ static int api_rx_disable(struct device *dev)
 	return uart_rx_disable(data->uart);
 }
 
+#if CONFIG_NRF_SW_LPUART_INT_DRIVEN
+
+static uint32_t int_driven_rd_available(struct lpuart_data *data)
+{
+	return data->int_driven.rxlen - data->int_driven.rxrd;
+}
+
+static void int_driven_rx_feed(struct device *dev, struct lpuart_data *data)
+{
+	int err;
+
+	data->int_driven.rxlen = 0;
+	data->int_driven.rxrd = 0;
+	err = api_rx_buf_rsp(dev, data->int_driven.rxbuf,
+			     sizeof(data->int_driven.rxbuf));
+	__ASSERT_NO_MSG(err >= 0);
+}
+
+static void int_driven_evt_handler(struct device *uart,
```

Comment:
Perhaps naming this parameter `lpuart` could help avoid confusion with the same parameter in `uart_callback()` that points to a different thing.